tests/unit/application/use-cases/log-event.use-case.test.ts

```diff
@@ -0,0 +1,77 @@
+import { MockEventsRepository } from '@/infrastructure/repositories/events.repository.mock';
+import { describe, it, beforeEach, expect } from 'vitest';
+import { getInjection } from '@/di/container';
+
+const logEventUseCase = getInjection('ILogEventUseCase');
+const eventsRepo = getInjection('IEventsRepository') as MockEventsRepository;
+
+describe('logEventUseCase', () => {
+    beforeEach(() => {
+        eventsRepo.clear();
+    });
+
+    it('creates and stores an event successfully', async () => {
+        const now = new Date('2025-10-21T12:00:00Z');
+        const result = await logEventUseCase.execute({
+            userId: 1,
+            sessionId: 100,
+            actionName: 'SOLVE_PROBLEM',
+            loggedAt: now,
+            payload: '{"attempt":1}',
+        });
+
+        expect(result).toMatchObject({
+            id: 1,
+            userId: 1,
+            sessionId: 100,
+            actionName: 'SOLVE_PROBLEM',
+            loggedAt: now,
+        });
```

Comment:
As the other test-cases checks the timestamp handling, this test-case should not check it (if this test-case checks it, there is no need for the last test-case).

```suggestion
    it('creates and stores an event successfully', async () => {
        const result = await logEventUseCase.execute({
            userId: 1,
            sessionId: 100,
            actionName: 'SOLVE_PROBLEM',
            payload: '{"attempt":1}',
        });

        expect(result).toMatchObject({
            id: 1,
            userId: 1,
            sessionId: 100,
            actionName: 'SOLVE_PROBLEM',
        });
```